### PR TITLE
chore: evaluation remarks corrected

### PR DESCRIPTION
**Pull request by asmie**
This PR updates the repository to address evaluation remarks.

1. Docker is building now correctly. Removed the --runtime-benchmarks flag as PoC code does not support benchmarking.
2. Although benchmarking is not supported, now code is at least building correctly with that feature enabled.
3. `cargo test` now passes.
4. Corrected Testing Guide for RPC testing. Users need to run the node with the --dev chain.
5. Made various clarifications and corrections to XCM testing and benchmarking.
6. Added missing files.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -12,10 +12,10 @@ RUN git clone https://github.com/paritytech/polkadot-sdk.git && cd polkadot-sdk
 # Delete below three lines when polka-storage repo will become public
 WORKDIR /usr/src/polka-storage
 COPY . .
-RUN cargo build --release --features runtime-benchmarks
+RUN cargo build --release
 
-# Uncomment when polka-storage repo will become public
-#RUN git clone https://github.com/eigerco/polka-storage.git && cd polka-storage && cargo build --release --features runtime-benchmarks
+# Uncomment when polka-storage repo will become public and available for cloning for everyone
+#RUN git clone https://github.com/eigerco/polkadot-native-storage polka-storage && cd polka-storage && cargo build --release --features runtime-benchmarks
```
Comment (kylegranger) [inline Dockerfile:19]: Is there any reason to rename the repo to `polka-storage`?  I would prefer not to, but maybe you have reasons?  E.g., do many scripts depend on the name `polka-storage`?

File: README.md [hunk 2]
```diff
@@ -11,9 +11,9 @@ Repository structure:
 - `scripts` - contains the scripts used to build or test the project
 
 ## Building the node
-Download the `polka-storage` parachain repository:
+Download the `polka-native-storage` parachain repository:
 ```bash
-git clone https://github.com/eigerco/polka-storage.git
+git clone https://github.com/eigerco/polkadot-native-storage polka-storage
 ```
```
Comment (kylegranger) [inline README.md:17]: same for renaming here

File: doc/testing_guide.md [hunk 3]
```diff
@@ -78,7 +78,7 @@ yarn start
 ### Local parachain (collator)
 Download the `polka-storage` parachain repository:
 ```bash
```
Comment (kylegranger) [inline doc/testing_guide.md:80]: Download the `polkadot-native-storage`...

File: doc/testing_guide.md [hunk 4]
```diff
@@ -78,7 +78,7 @@ yarn start
...
-git clone https://github.com/eigerco/polka-storage.git
+git clone https://github.com/eigerco/polkadot-native-storage polka-storage
```
Comment (kylegranger) [inline doc/testing_guide.md:81]: and here.

File: Dockerfile [hunk 5]
```diff
@@ -12,10 +12,10 @@ RUN git clone https://github.com/paritytech/polkadot-sdk.git && cd polkadot-sdk
 # Delete below three lines when polka-storage repo will become public
 WORKDIR /usr/src/polka-storage
 COPY . .
-RUN cargo build --release --features runtime-benchmarks
+RUN cargo build --release
 
-# Uncomment when polka-storage repo will become public
-#RUN git clone https://github.com/eigerco/polka-storage.git && cd polka-storage && cargo build --release --features runtime-benchmarks
+# Uncomment when polka-storage repo will become public and available for cloning for everyone
```
Comment (kylegranger) [inline Dockerfile]: Uncomment when polka-native-storage repo

File: doc/testing_guide.md [hunk 6]
```diff
@@ -36,11 +36,11 @@ Required tools:
...
 Users should also have a basic understanding of the Substrate framework and the Polkadot ecosystem.
 
-The requirements we can test with the local setup and provided PoC code:
+The requirements we can test or show with the local setup and provided PoC code:
 - parachain registration to local relay chain;
 - parachain block production;
-- XCM (Cross-Chain Messaging) - sending messages between parachains using sudo pallet and ping pallet;
-- visibility of the Filecoin actors as pallets in the local parachain;
+- XCM (Cross-Chain Messaging) - sending messages between parachains using official sudo pallet;
+- visibility of the Filecoin actors as pallets in the local node;
```
Comment (kylegranger) [inline doc/testing_guide.md:43]: using `the` official sudo pallet

File: pallets/pns/src/mock.rs [hunk 7]
```diff
@@ -13,7 +13,7 @@ frame_support::construct_runtime!(
     pub enum Test
     {
         System: frame_system::{Pallet, Call, Config<T>, Storage, Event<T>},
-        MinerModule: crate::{Pallet, Call, Storage, Event<T>},
+        PnsModule: crate::{Pallet, Call, Storage, Event<T>},
     }
```
Comment (kylegranger) [inline pallets/pns/src/mock.rs:17]: nice!  I like the name.

**Review by kylegranger — APPROVED** [review]
LGTM 👍

File: Dockerfile (hunk 1 above)
Comment (asmie) [reply Dockerfile:19]: The one and only reason is maintainability.
It should be easier to keep the main repo and the public one aligned when there will be only one line to change (and the rest of the script can use `polka-storage`). But if it's somehow unwanted - no problem, I'll change it 👍

Comment (kylegranger) [reply Dockerfile:19]: Understood.  Then let's keep it.

File: README.md (hunk 2 above)
Comment (asmie) [reply README.md:17]: Same as the first one.

File: doc/testing_guide.md (hunk 6 above)
Comment (asmie) [reply doc/testing_guide.md:43]: Corrected.

File: Dockerfile (hunk 5 above)
Comment (asmie) [reply Dockerfile]: Corrected.

File: doc/testing_guide.md (hunk 4 above)
Comment (asmie) [reply doc/testing_guide.md:81]: Let's leave it as the first message says.

File: doc/testing_guide.md (hunk 3 above)
Comment (asmie) [reply doc/testing_guide.md:80]: Corrected.

File: README.md (hunk 2 above)
Comment (kylegranger) [reply README.md:17]: 👍

File: doc/testing_guide.md (hunk 4 above)
Comment (kylegranger) [reply doc/testing_guide.md:81]: 👍

**Pull request merged**